analysis/figures/figure_subgroups.R

```diff
@@ -0,0 +1,242 @@
+    "Age group - Vaccinated",
+    "Age group - Unvaccinated",
+    "Ethnicity - Pre-vaccination",
+    "Ethnicity - Vaccinated",
+    "Ethnicity - Unvaccinated",
+    "Prior history of event - Pre-vaccination",
+    "Prior history of event - Vaccinated",
+    "Prior history of event - Unvaccinated",
+    "Prior GI operations - Pre-vaccination",
+    "Prior GI operations - Vaccinated",
+    "Prior GI operations - Unvaccinated",
+    "Sex - Pre-vaccination",
+    "Sex - Vaccinated",
+    "Sex - Unvaccinated"
+  )
+)
+
+names <- c(
+  `Age group - Pre-vaccination` = "Pre-vaccination",
+  `Age group - Vaccinated` = "Vaccinated\nAge group",
+  `Age group - Unvaccinated` = "Unvaccinated",
+  `Ethnicity - Pre-vaccination` = "",
+  `Ethnicity - Vaccinated` = "Ethnicity",
+  `Ethnicity - Unvaccinated` = "",
+  `Prior history of event - Pre-vaccination` = "",
+  `Prior history of event - Vaccinated` = "Prior history of event",
+  `Prior history of event - Unvaccinated` = "",
+  `Prior GI operations - Pre-vaccination` = "",
+  `Prior GI operations - Vaccinated` = "Prior GI operations",
+  `Prior GI operations - Unvaccinated` = "",
+  `Sex - Pre-vaccination` = "",
+  `Sex - Vaccinated` = "Sex",
+  `Sex - Unvaccinated` = ""
+)
+
+names_2col <- c(
+  `Age group - Pre-vaccination` = "Pre-vaccination\nAge group",
+  `Age group - Vaccinated` = "Vaccinated",
+  `Age group - Unvaccinated` = "Unvaccinated",
+  `Ethnicity - Pre-vaccination` = "Ethnicity",
+  `Ethnicity - Vaccinated` = "",
+  `Ethnicity - Unvaccinated` = "",
+  `Prior history of event - Pre-vaccination` = "Prior history of event",
+  `Prior history of event - Vaccinated` = "",
+  `Prior history of event - Unvaccinated` = "",
+  `Prior GI operations - Pre-vaccination` = "Prior GI operations",
+  `Prior GI operations - Vaccinated` = "",
+  `Prior GI operations - Unvaccinated` = "",
+  `Sex - Pre-vaccination` = "Sex",
+  `Sex - Vaccinated` = "",
+  `Sex - Unvaccinated` = ""
+)
+
+for (outcome_name in unique(subgroups$outcome)) {
+  
+
+  df <- subgroups %>% filter(outcome == outcome_name)
+
+  analysis_labels_levels <- unique(df$analysis_labels)
+  df$analysis_labels <- factor(df$analysis_labels, levels = analysis_labels_levels)
+
+  colour_levels <- unique(df$colour)
+  df$colour <- factor(df$colour, levels = colour_levels)
+
+  pd <- position_dodge(width = 0.5)
+  p <- ggplot(df, aes(x = outcome_time_median / 7, y = hr, color = colour)) +
+    geom_line() +
+    geom_point(size = 2, position = pd) +
+    geom_hline(mapping = aes(yintercept = 1), colour = "#A9A9A9") +
+    geom_errorbar(
+      size = 1.2,
+      mapping = aes(
+        ymin = ifelse(conf_low < 0.25, 0.25, conf_low),
+        ymax = ifelse(conf_high > 64, 64, conf_high),
+        width = 0
+      ),
+      position = pd
+    ) +
+    scale_x_continuous(breaks = seq(0, max(df$outcome_time_median) / 7, 4)) +
+    scale_y_continuous(
+      lim = c(0.25, 64),
+      breaks = c(0.25, 0.5, 1, 2, 4, 8, 16, 64),
```

Comment:
Again, I suggest to add 32 here. Otherwise it looks like a gap in the y-scale.
```suggestion
      breaks = c(0.25, 0.5, 1, 2, 4, 8, 16, 32, 64),
```